### ISSUE TITLE
Do not validate empty fields that is not required.

### ISSUE THREAD

**Issue by hovsater** [posabsolute/jQuery-Validation-Engine#430]
I went through the code for field validation and I think I discovered a bug. Fields that are not required by the means of `validate[required]` and have an empty value should not trigger validation.

Currently we have the following:

``` javascript
// If the rules required is not added, an empty field is not validated
if(!required && field.val() == "" && !promptText) options.isError = false;
```

I suppose that this if-statement actually is intended to prevent validation for this kind of situations. The problem with the current implementation is that we have an if-statement just before this:

``` javascript
if (errorMsg !== undefined)  {
  promptText += errorMsg + "<br/>";
  options.isError = true;
}
```

this if-statement will actually set the `promptText` to a string size greater than zero if an error message exists, resulting in validating fields that actually should not be validated. An error message might be present due to a custom validation made by a user although the field itself don't have a `required` validation nor a value.

This shouldn't be to hard to fix. Either I do a pull request with a fix for this or you just implement it yourself. It doesn't matter.
